analysis/src/lib.rs

```diff
@@ -1,26 +1,47 @@
 //! This crate contains a number of analyses that can be run on dependencies
 
+use anyhow::Result;
 use guppy::graph::PackageMetadata;
 use tabled::Tabled;
 
+mod cratesio;
+
 #[derive(Tabled)]
 pub struct DependencyReport<'a> {
     pub name: &'a str,
     pub has_build_script: bool,
+    pub hosted_on_cratesio: bool,
+    pub cratesio_downloads: u64,
+    pub cratesio_dependents: u64,
 }
 
 pub struct DependencyAnalyzer;
 
 impl DependencyAnalyzer {
-    pub fn analyze_dep<'a>(&self, package: &PackageMetadata<'a>) -> DependencyReport<'a> {
+    pub fn analyze_dep<'a>(&self, package: &PackageMetadata<'a>) -> Result<DependencyReport<'a>> {
         let name = package.name();
         let has_build_script = package.has_build_script();
 
-        // Add more analyses here
+        // Crates.io analysis
+        let hosted_on_cratesio = package.source().is_crates_io();
+        let mut cratesio_downloads = 0;
+        let mut cratesio_dependents = 0;
+
+        if hosted_on_cratesio {
+            let cratesio_report = cratesio::CratesioAnalyzer::new()?;
+            let cratesio_report = cratesio_report.analyze_cratesio(package)?;
+            cratesio_downloads = cratesio_report.downloads;
+            cratesio_dependents = cratesio_report.dependents;
+        }
```

Comment:
As I mentioned above, if we do this check inside the `analyze_cratesio` function we'll be able to simplify this code.